### PR TITLE
Replace aggregate_args

### LINKED ISSUE #147: Missing arguments
@gwaygenomics I could be wrong or I am missing something obvious but it looks like `aggregate_args` is not being passed to `aggregate_compartment` when the function is called within `aggregate_profiles`. 

https://github.com/cytomining/pycytominer/blob/4f37e019918256111933aed4e3b9afb50055b4a1/pycytominer/cyto_utils/cells.py#L621-L671

I am currently using `aggregate_args` to set `operation` to `mean` or `median` for `aggregate` in the profiling recipe but I found that `aggregate` always uses the default `operation`.

### PR DESCRIPTION
**Pull request by niranjchandrasekaran**
_modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_

# Description

I replace `aggregate_args` in `aggregate_compartments` and `aggregate_profiles` as described in https://github.com/cytomining/pycytominer/issues/147#issuecomment-862457682. 

closes #147

## What is the nature of your change?

- [x] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] This change requires a documentation update.

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: pycytominer/aggregate.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def aggregate(
         DataFrame to group and aggregate.
     strata : list of str, default ["Metadata_Plate", "Metadata_Well"]
         Columns to groupby and aggregate.
-    features : list of str, default "all"
+    features : list of str, default "infer"
```
Comment (gwaygenomics) [inline pycytominer/aggregate.py:34]: 👍

File: pycytominer/cyto_utils/cells.py [hunk 2]
```diff
@@ -436,27 +437,21 @@ def aggregate_compartment(
...
 
         # Infering features is tricky with non-canonical data
-        if aggregate_args is None:
-            aggregate_args = {}
-            features = infer_cp_features(population_df, compartments=compartment)
-        elif "features" not in aggregate_args:
-            features = infer_cp_features(population_df, compartments=compartment)
-        elif aggregate_args["features"] == "infer":
-            features = infer_cp_features(population_df, compartments=compartment)
+        if self.features == "infer":
```
Comment (gwaygenomics) [inline pycytominer/cyto_utils/cells.py:440]: Ah, this is so much nicer... You can remove line 439 - it is no longer tricky!

**Review by gwaygenomics — APPROVED** [review]
Looks great

File: pycytominer/cyto_utils/cells.py (hunk 2 above)
Comment (niranjchandrasekaran) [reply pycytominer/cyto_utils/cells.py:439]: :) Done

**Comment by niranjchandrasekaran** [comment]
@gwaygenomics Probably a naive question - I don't see the check mark from Travis CI for this PR. Am I missing something?

**Comment by gwaygenomics** [comment]
oh weird, me neither. I wonder if it has to do with travis migrating from .org to .com

image

It also doesn't seem to exist... https://www.travis-ci.com/github/cytomining/pycytominer/builds

I think rather than trying to solve travis, we should just upgrade pycytominer to github actions. We use this in cytominer-eval already: https://github.com/cytomining/cytominer-eval/blob/master/.github/workflows/codecov.yml and it is quite easy to setup

**Comment by niranjchandrasekaran** [comment]
👍  I will create a new issue quoting your comment.

What do you suggest we do for this PR?

**Comment by gwaygenomics** [comment]
It's approved, so you can merge if you need this functionality asap.

Otherwise, I suggest we go about it the more "best practices" route:

1. Solve #150
2. Create some "dummy" commit to trigger the action
3. Merge once it's passed

**Comment by niranjchandrasekaran** [comment]
>Otherwise, I suggest we go about it the more "best practices" route:

Sounds good

**Comment by niranjchandrasekaran** [comment]
Yaay! It passed :) @gwaygenomics everything looks good?

**Comment by gwaygenomics** [comment]
LGTM 👍

**Pull request merged**